broadlink/__init__.py

```diff
@@ -14,66 +14,85 @@
 
 from .exceptions import check_error, exception
 
+
 def gendevice(devtype, host, mac, name=None, cloud=None):
     devices = {
-        sp1: [0],
```

Comment:
In a future where we no longer need to support Python 2, I can make this function 84x faster using the lru_cache decorator.
```python3
>>> timeit(without_lru_cache)
19.186042481000186

>>> timeit(with_lru_cache)
0.22894878300576238
```

Reply:
I think we should drop python2 support